File: PAV 1-TPI - CASOS DE USO version 1/Resueltos/Usuarios+SeleccionActBugs/App_BugTracker resuelto/App_BugTracker/GUILayer/Bugs/Frm_detalle_bug.vb
Public Class Frm_detalle_bug
    Public Sub mostrar_bug(ByVal oBugSelected As Bug)
        txt_nro_bug.Text = oBugSelected.id_bug.ToString
        txt_titulo.Text = oBugSelected.titulo
        txt_producto.Text = oBugSelected.n_producto
        txt_fec_alta.Text = oBugSelected.fecha_alta
        txt_descri.Text = oBugSelected.descripcion
        txt_estado.Text = oBugSelected.estado
        txt_prioridad.Text = oBugSelected.n_prioridad
        txt_criticidad.Text = oBugSelected.n_criticidad

        dgv_historiales.Rows.Clear()
        For Each oHistorialBug As HistorialBug In oBugSelected.historial
            dgv_historiales.Rows.Add(New String() {oHistorialBug.fecha.ToString("dd/MM/yyyy"), oHistorialBug.n_responsable, oHistorialBug.n_estado, oHistorialBug.n_asignado_a})
        Next
    End Sub

    Private Sub btn_aceptar_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btn_aceptar.Click
        Me.Close()
    End Sub

    Private Sub Frm_detalle_bug_Load(sender As System.Object, e As System.EventArgs) Handles MyBase.Load

    End Sub
End Class

File: Clases 2018/Clases-C#/ClasePractica6_2018 (consulta simple bugs en capas)/C#_TPS_BugTracker/TPS_BugTracker/GUILayer/frmDetalleBug.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using TPS_BugTracker.BusinessLayer.Entities;

namespace TPS_BugTracker
{
    public partial class frmDetalleBug : Form
    {
        public frmDetalleBug()
        {
            InitializeComponent();
        }

        internal void mostrarBug(Bug oBugSelected)
        {
            txtNroBug.Text = oBugSelected.id_bug.ToString();
            txtTitulo.Text = oBugSelected.titulo;
            txtProducto.Text = oBugSelected.n_producto;
            txtFechaAlta.Text = oBugSelected.fecha_alta.ToString();
            txtDescripcion.Text = oBugSelected.descripcion;
            txtEstado.Text = oBugSelected.estado;
            txtPrioridad.Text = oBugSelected.n_prioridad;
            txtCriticidad.Text = oBugSelected.n_criticidad;

            dgv_historiales.Rows.Clear();
          //  foreach (ElementoBug oBug in oBugSelected.historial)
            //    dgv_historiales.Rows.Add(new string[] { oBug.fecha.ToString("dd/MM/yyyy"), oBug.n_responsable, oBug.n_estado, oBug.n_asignado_a });
        }
    }
}
